test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -712,6 +715,158 @@ func TestMultiplePolicyBindings(t *testing.T) {
+      name: "demo-policy.example.com"
+    spec:
+      failurePolicy: Fail
+      matchConstraints:
+        resourceRules:
+        - apiGroups:   ["*"]
+          apiVersions: ["*"]
+          operations:  ["*"]
+          resources:   ["deployments"]
+      validations:
+        - expression: "!has(object.spec.paused)"    
+`), 8192).Decode(&vap))
+	require.NoError(t, yaml.NewYAMLOrJSONDecoder(bytes.NewBufferString(`
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "demo-binding-test.example.com"
+spec:
+  policyName: "demo-policy.example.com"
+  validationActions: [Deny]
+`), 8192).Decode(&vab))
+
+	opts := metav1.ApplyOptions{
+		FieldManager: "manager",
+	}
+
+	etcd.CreateTestCRDs(t, apiextensionsclientset.NewForConfigOrDie(server.ClientConfig), false, crd)
+
+	_, e := dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "admissionregistration.k8s.io",
+		Version:  "v1beta1",
+		Resource: "validatingadmissionpolicies",
+	}).Apply(context.TODO(), "demo-policy.example.com", &vap, opts)
+	require.NoError(t, e)
+	_, e = dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "admissionregistration.k8s.io",
+		Version:  "v1beta1",
+		Resource: "validatingadmissionpolicybindings",
+	}).Apply(context.TODO(), "demo-binding-test.example.com", &vab, opts)
+	require.NoError(t, e)
+
+	var deploy unstructured.Unstructured
+	var deploy2 unstructured.Unstructured
+
+	require.NoError(t, yaml.NewYAMLOrJSONDecoder(bytes.NewBufferString(`
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: nginx-deployment
+    spec:
+      paused: false
+      replicas: 3
+      selector:
+        matchLabels:
+          app: nginx
+      template:
+        metadata:
+          labels:
+            app: nginx
+        spec:
+          containers:
+            - image: nginx
+              name: nginx
+`), 8192).Decode(&deploy))
+	require.NoError(t, yaml.NewYAMLOrJSONDecoder(bytes.NewBufferString(`
+apiVersion: stable.example.com/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  paused: false
+`), 8192).Decode(&deploy2))
+	time.Sleep(3 * time.Second)
+
+	// Apply a ConfigMap with `false` for non-pointer omitempty field.
+	// It should be converted into unstructured and `has` returns false
+	// And therefore pass the admission policy
+	_, e = dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}).Namespace("default").Apply(context.TODO(), "nginx-deployment", &deploy, opts)
+	require.NoError(t, e)
+
+	// Apply a CRD ConfigMap with `false` for non-pointer omitempty field
+	// It should retain knowledge that `false` was specified
+	_, e = dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "stable.example.com",
+		Version:  "v1",
+		Resource: "deployments",
+	}).Namespace("default").Apply(context.TODO(), "nginx-deployment", &deploy2, opts)
+	require.NoError(t, e)
```

Comment:
CRD with `paused: false` evaluates has(spec.paused) in the schema to `true`, even though the schema is the same.

Reply:
I think this issue is slightly different than the one reported in https://github.com/kubernetes/kubernetes/issues/120226

I *think* that issue was saying a CRD stanza defining an object with specific `properties` and one defining an object as a map (`additionalProperties`) behaved differently w.r.t. `has` for the same JSON input